### DISCUSSION TITLE
Continuous Task Sync Questions (prior beta release)

### DISCUSSION THREAD

**Discussion by FlohGro-dev** [FlohGro-dev/Craftist#42] [Q&A]
Hi there I would like to ask for your feedback regarding continuous sync of tasks.
In issue [#34](https://github.com/FlohGro-dev/Craftist/issues/34) it's requested to continuously sync the tasks.

I experimented a little bit with it but quickly landed on a few questions.
First I refactored the updating of blocks which is no much cleaner and now blocks are removed and re-added anymore. This results in a much cleaner experience of that feature.

I prototyped a continuous sync for the tasks (this is not finished and needs a lot of cleanup but..) - I could enable / disable it with the same butten which changes its description when it's enabled / disabled.
First question seems to be easy:

**what time period do you expect for syncing?**
- E.g. every 5 seconds?
- faster / slower?

Next: 
Craft shows a small notification when blocks were updated (similar to the banner when you copy a block)
This banner can't be disabled
**would it distract you more than help you when the banner will be displayed on every sync (if yes this is a dealbreaker for that feature)**

Last (for now):
It's currently not possible to detect when the user switches a document & sometimes the API does not return the current open page correctly to the eXtension (especially when you switch quickly between them)
**is it fine for you that an enabled continuous sync will not be stopped when you switch documents?**

Thanks for your thoughts!

**Comment by Elwood4415** [comment]
Thanks for asking for feedback. Here are my responses (as of right now):

1. Continuous sync would be nice but, for me at least, I don't think it needs as quick as every 5 seconds especially if that is going to be taxing systems somewhere. If more often is a problem, I would be fine with once an hour or so.
2. I am not entirely sure what you mean by the banner, but I think that would be fine.
3. Again, not sure of all the implications of this, but I think it is fine to continue to sync tasks even when we change documents.

I appreciate your work on this and look forward to seeing the results!

**Reply by FlohGro-dev** [reply]
thanks for the response really valuable for me.

regarding 2): I'm talking about this banner:
![CleanShot 2022-03-25 at 19 52 57](https://user-images.githubusercontent.com/13785667/160183861-30b4078c-8abf-4683-9cbc-2c9017f12bab.png)

**Comment by Elwood4415** [comment]
I am fine with that banner. I was thinking you meant some kind of persistent one.

**Reply by FlohGro-dev** [reply]
no :) this will just be presented everytime the sync updates the blocks in the document 👍

**Comment by FlohGro-dev** [comment]
thanks for the feedback, further discussion of the beta release in [this discussion](https://github.com/FlohGro-dev/Craftist/discussions/45)
